blog-server-services/src/impls/social_service.rs

```diff
@@ -0,0 +1,96 @@
+use blog_generic::events::SubscriptionStateChanged;
+use rbatis::rbdc::rt::tokio;
+use screw_components::dyn_result::{DError, DResult};
+use std::sync::Arc;
+
+use crate::traits::author_service::{Author, AuthorService, BaseMinimalAuthor};
+use crate::traits::event_bus_service::EventBusService;
+use crate::traits::social_service::{SocialId, SocialService as SocialServiceTrait};
+
+pub fn create_social_service(
+    author_service: Arc<Box<dyn AuthorService>>,
+    event_bus_service: Arc<Box<dyn EventBusService>>,
+) -> Box<dyn SocialServiceTrait> {
+    Box::new(SocialService {
+        author_service,
+        event_bus_service,
+    })
+}
+struct SocialService {
+    author_service: Arc<Box<dyn AuthorService>>,
+    event_bus_service: Arc<Box<dyn EventBusService>>,
+}
+
+#[async_trait]
+impl SocialServiceTrait for SocialService {
+    async fn set_subscribe_for_author(&self, author: &Author, subscribe: &u8) -> DResult<()> {
+        if let Some(telegram_id) = author.base.telegram_id {
+            let event = SubscriptionStateChanged {
+                blog_user_id: author.id,
+                user_telegram_id: telegram_id,
+                new_state: *subscribe,
+            };
+
+            let event_bus_service = self.event_bus_service.clone();
+            tokio::spawn(async move { event_bus_service.publish(event).await });
+        } else {
+            Err(DError::from("not supported for current author"))?
+        }
+
+        self.author_service
+            .set_author_subscription_by_id(&author.id, &subscribe)
+            .await
+    }
+    async fn process_auth_by_id(
```

Comment:
i believe we can split it further to make code look more understandable:

get_author_by_social
if found 
    update (handles logic with overriding)
    return

insert new


not really functional way though *thinking*

Reply:
will add to https://github.com/users/tikitko/projects/2?pane=issue&itemId=45347228